homeassistant/components/scrape/config_flow.py

```diff
@@ -95,26 +95,31 @@
     vol.Optional(CONF_ENCODING, default=DEFAULT_ENCODING): TextSelector(),
 }
 
+NONE_SENTINEL = "none"
+
 SENSOR_SETUP = {
     vol.Required(CONF_SELECT): TextSelector(),
     vol.Optional(CONF_INDEX, default=0): NumberSelector(
         NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
     ),
     vol.Optional(CONF_ATTRIBUTE): TextSelector(),
     vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
-    vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
+    # Note: we set default to ensure that frontend does not omit the result
+    vol.Optional(CONF_DEVICE_CLASS, default=NONE_SENTINEL): SelectSelector(
         SelectSelectorConfig(
-            options=[cls.value for cls in SensorDeviceClass],
+            options=[NONE_SENTINEL] + [cls.value for cls in SensorDeviceClass],
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
-    vol.Optional(CONF_STATE_CLASS): SelectSelector(
+    # Note: we set default to ensure that frontend does not omit the result
+    vol.Optional(CONF_STATE_CLASS, default=NONE_SENTINEL): SelectSelector(
         SelectSelectorConfig(
-            options=[cls.value for cls in SensorStateClass],
+            options=[NONE_SENTINEL] + [cls.value for cls in SensorStateClass],
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
-    vol.Optional(CONF_UNIT_OF_MEASUREMENT): SelectSelector(
+    # Note: we set default to ensure that frontend does not omit the result
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT, default=""): SelectSelector(
```

Comment:
I have not used the sentinel for unit of measurement, as it can be any string (`""` is a valid unit)
But we could adjust it if the reviewers feel it is needed...

Reply:
I think we should allow the user to remove the unit of measurement, so add the "none" option here too so the user can select it